Paco/src/com/pacoapp/paco/model/ExperimentProviderUtil.java

```diff
@@ -747,10 +773,15 @@ public void loadEventsForExperiment(Experiment experiment) {
   }
 
   public List<Event> loadEventsForExperimentByServerId(Long serverId) {
+	//TODO Should be calling the below method, with some default value ??
     return findEventsBy(EventColumns.EXPERIMENT_SERVER_ID + " = " + Long.toString(serverId),
         EventColumns._ID +" DESC");
   }
 
+  public List<Event> loadEventsForExperimentByServerId(Long serverId, int noOfRecords) {
+	    return findEventsBy(EventColumns.EXPERIMENT_SERVER_ID + " = " + Long.toString(serverId),
+	        EventColumns._ID +" DESC", noOfRecords);
```

Comment:
I think it will be increasing unless we change the settings to do otherwise. The change here was just to add the noOfRecords parameter. With respect to the rest of the code it remains the same. I have modified the code slightly to make it visible.